src/core/thread/network_data_local.cpp

```diff
@@ -69,6 +68,14 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
     // of an IEEE 802.15.4 interface MUST have a length of 64 bits.
     VerifyOrExit(!aConfig.mSlaac || aConfig.mPrefix.mLength == OT_IP6_PREFIX_BITSIZE, error = kErrorInvalidArgs);
 
+exit:
```

Comment:
Should we validate more:

- VerifyOrExit((aPrefix.GetLength() > 0) && aPrefix.IsValid(), error = kErrorInvalidArgs);
- Verify `Preference` is valid
- Verify Prefix contains Mesh-local Prefix
- Verify Network Data contains enough space

I think the original code intended to make sure that the prefix can be added (enough space in Network Data). 

@librasungirl